conduit/src/Data/Conduit/Internal/Conduit.hs

```diff
@@ -720,6 +723,38 @@ connect :: Monad m
         -> m r
 connect = ($$)
 
+-- | Split a conduit into head and tail.
+--
+-- Note that you have to 'sealConduitT' it first.
+--
+-- Since 1.3.3
+uncons :: forall m o conduit. (Monad m, conduit ~ SealedConduitT () o m ())
+       => conduit -> m (Maybe (o, conduit))
+uncons (SealedConduitT p) = fmap (fmap (second SealedConduitT)) $ go p
+  where
+    -- This function is the same as @Pipe.uncons@ but it ignores leftovers.
+    go (HaveOutput p o) = pure $ Just (o, p)
+    go (NeedInput _ c) = go $ c ()
+    go (Done ()) = pure Nothing
+    go (PipeM mp) = mp >>= go
+    go (Leftover p ()) = go p
+
+-- | Split a coundit into head and tail or return its result if it is done.
+--
+-- Note that you have to 'sealConduitT' it first.
+--
+-- Since 1.3.3
+unconsE :: forall m o r conduit. (Monad m, conduit ~ SealedConduitT () o m r)
```

Comment:
The distinction between `uncons` and `unconsE` doesn't seem to follow the standard naming scheme of "stream of chunks" vs "stream of values" found in the rest of the library

Reply:
Oops, I probably took this from some other streaming library...